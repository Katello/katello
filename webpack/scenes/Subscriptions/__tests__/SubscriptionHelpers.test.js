import {
  quantitiesRequestSuccessResponse,
  loadQuantitiesSuccessActionPayload,
} from './subscriptions.fixtures';
import {
  filterRHSubscriptions,
  manifestExists,
  selectSubscriptionsQuantitiesFromResponse,
} from '../SubscriptionHelpers';

describe('Subscription helper', () => {
  it('should filter redhat subscriptions', () => {
    const subscriptions = [
      {
        key: 'sub-1',
        available: 0,
      },
      {
        key: 'sub-2',
        available: 4,
      },
      {
        key: 'sub-3',
        available: -5,
      },
      {
        key: 'sub-4',
        available: 100,
      },
    ];

    const filteredSubscriptions = filterRHSubscriptions(subscriptions);

    expect(filteredSubscriptions).toMatchSnapshot();
  });

  it('should check if manifest exists in an organization', () => {
    const upstreamConsumer = 'some-upstream-consumer';

    expect(manifestExists({
      owner_details: { upstreamConsumer },
    })).toBe(upstreamConsumer);

    expect(manifestExists({})).toBeFalsy();
  });

  it('should select subscriptions-quantities from api response', () => {
    const quantities =
      selectSubscriptionsQuantitiesFromResponse(quantitiesRequestSuccessResponse);

    expect(quantities).toEqual(loadQuantitiesSuccessActionPayload);
  });
});
