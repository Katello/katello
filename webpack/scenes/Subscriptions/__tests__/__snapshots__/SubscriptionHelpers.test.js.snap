// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Subscription helper should filter redhat subscriptions 1`] = `
Array [
  Object {
    "available": 0,
    "key": "sub-1",
  },
  Object {
    "available": 4,
    "key": "sub-2",
  },
  Object {
    "available": 100,
    "key": "sub-4",
  },
]
`;
